Sources/ConfidenceProvider/Apply/FlagApplierWithRetries.swift

```diff
@@ -21,69 +23,86 @@ final class FlagApplierWithRetries: FlagApplier {
         self.storage = storage
         self.httpClient = httpClient
         self.options = options
-        self.cacheDataInteractor = cacheDataInteractor ?? CacheDataInteractor(storage: storage)
+
+        // Read/Write operations to/from the file are not protected by the cacheDataActor
+        // It's important to not write the file before this operation is completed (reading from file)
+        let storedData = try? storage.load(defaultValue: CacheData.empty())
```

Comment:
I'm not sure I understand this comment - read/write thread safety is handled by the `Storage` object, does that not protect us from writing before reading, or is there something more to this?

Reply:
I believe this comment is not relevant anymore. I have removed storage from cacheDataInteractor.

Reply:
Removed it ✅